activities/Gwen.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+"""
+Instructions
+
+- Functions intended to run for each scan
+  need to start with `run_`, e.g., `run_slope`.
+
+- Do not modify the parameters of the `run_` function
+  unless you know what you are doing.
+  See optional parameters at:
+  https://github.com/tangible-landscape/grass-tangible-landscape/wiki/Running-analyses-and-developing-workflows#python-workflows
+
+- All gs.run_command/read_command/write_command/parse_command
+  need to be passed *env* parameter like this: `(..., env=env)`.
+"""
+
+import os
+
+import grass.script as gs
+
+
+def run_function_with_points(scanned_elev, env, points=None, **kwargs):
+    """Doesn't do anything, except loading points from a vector map to Python
+
+    If *points* is provided, the function assumes it is name of an existing vector map.
+    This is used during testing.
+    If *points* is not provided, the function assumes it runs in Tangible Landscape.
+    """
+    if not points:
+        # If there are no points, ask Tangible Landscape to generate points from
+        # a change in the surface.
+        points = "points"
+        import analyses
+
+        analyses.change_detection(
+            "scan_saved",
+            scanned_elev,
+            points,
+            height_threshold=[10, 100],
+            cells_threshold=[5, 50],
+            add=True,
+            max_detected=5,
+            debug=True,
+            env=env,
+        )
+
+    # generate random point for focal garden
+    gs.run_command("v.random", output="focal_garden", npoints=1, seed=3)
+
+    # calculate the cost surface
+    gs.run_command(
+        "r.cost",
+        input="cfactorgrow_1m",
+        output="cost_surface",
+        start_points="focal_garden",
+        stop_points=points,
+    )
+
+    # calculate connectivity metric
+    data = gs.read_command("r.what", map="cost_surface", points=points)
+    data_lines = data.splitlines()
+    data = data_lines.split("|")
```

Comment:
From continuous integration failure:
```
Traceback (most recent call last):
  File "/home/runner/work/gis714-2024-tangible-landscape/gis714-2024-tangible-landscape/activities/Gwen.py", line 111, in <module>
    main()
  File "/home/runner/work/gis714-2024-tangible-landscape/gis714-2024-tangible-landscape/activities/Gwen.py", line 105, in main
    run_function_with_points(
  File "/home/runner/work/gis714-2024-tangible-landscape/gis714-2024-tangible-landscape/activities/Gwen.py", line 63, in run_function_with_points
    data = data_lines.split("|")
AttributeError: 'list' object has no attribute 'split'
```
Splitlines() returns list, not string, so you have to iterate through the list and call split on each line.